### ISSUE TITLE
PLIC on SAIL model

### ISSUE THREAD

**Issue by ptprasanna** [riscv-admin/dev-partners#34]
### Technical Group

PLIC

### ratification-pkg

PLIC

### Technical Liaison

Prasanna

### Task Category

SAIL

### Task Sub Category

- [ ] gcc
- [ ] binutils
- [ ] gdb
- [ ] intrinsics
- [ ] Java
- [ ] KVM
- [ ] ld
- [ ] llvm
- [ ] Linux kernel
- [ ] QEMU
- [ ] Spike

### Ratification Target

2022

### Statement of Work (SOW)

**Component names:**

**Requirements:**
Developing PLIC extension of Riscv  Specification, on Sail model using sail programming language. By implementing memory mapped registers on sail for handling the externally generated interrupts (platform generated interrupts).

**Deliverables:**
  - Component_1 and changes expected.
  - Component_2 and changes expected.
  ...

**Acceptance Criteria:**
  - Component_1 and measured result.
  - Component_2 and measured result.
  ...

**Projected timeframe:** (best guess date)

**SOW Signoffs:**
  - Task group liaison sign-off
  - Development partner sign-off
  - Golden Model SIG sign-off

### Waiver

- [ ] Freeze
- [ ] Ratification

**Comment by jjscheel** [comment]
Hi, @ptprasanna, I've added the SOW template to your first comment and moved your text into the requirements section.  Let's edit it there as-needed.

Thanks!

**Comment by jrtc27** [comment]
What's the point of having a PLIC without any devices attached to it that can generate interrupts?

**Comment by jjscheel** [comment]
Today's DevPartners meeting suggested that this work is on-going to create a PLIC infrastructure in SAIL and it will include a basic framework for testing interrupts.

The open question for the DevPartner's meeting is to determine how much to track here so as not to duplicate work.  At this time, I view this item as FYI for us.

**Comment by jrtc27** [comment]
NB: Bill doesn't review the code, the community does

**Comment by jjscheel** [comment]
@jrtc27, indeed.  I suspect that because the person is new, they're actually "pre-reviewing" the code with Bill.  Everyone understands that the community will review the code, comment, and ultimately accept the PR.

**Comment by jrtc27** [comment]
> @jrtc27, indeed. I suspect that because the person is new, they're actually "pre-reviewing" the code with Bill. Everyone understands that the community will review the code, comment, and ultimately accept the PR.

Ack

**Comment by jjscheel** [comment]
@billmcspadden-riscv, if you could provide and outlook on when a review of the code might be expected, that would help @ptprasanna.  Thanks!!!

**Comment by jjscheel** [comment]
@billmcspadden-riscv promises feedback this week.  Thanks, Bill!!!

**Comment by jjscheel** [comment]
@ptprasanna, please see the email to your gmail account on Bill's needs.

**Comment by jjscheel** [comment]
@ptprasanna, I would like an update in the issue, by the next meeting on December 12, 2023, please.

**Comment by jjscheel** [comment]
@anuani21, can you see if there's any status here.  @billmcspadden-riscv, is this the issue stalled behind the test framework work?

**Comment by jjscheel** [comment]
Thanks, @anuani21.  Please post when you have information.

**Comment by jjscheel** [comment]
Per @anuani21, the developer working on this has been pulled away with other activities.  Anusha can we please get an outlook for when they might be back?

Additionally, it would be helpful to add the developer as an assignee to the issue.  Are they in Groups.IO as a RISC-V member?  If not, we should get them there too so as to receive the mailing list information for the group.

**Comment by anuani21** [comment]
++saravanan

**Comment by jjscheel** [comment]
@ptprasanna, @anuani21, would you kindly provide status for the August 20 meeting on this item.  Thanks!

**Comment by JAYANTH-IITM** [comment]
Hi Sir ,
My Mail ID : jayanth.r0202@gmail.com
How may I get in touch with you sir

**Comment by jjscheel** [comment]
Thanks, Jayanth.  I've asked @billmcspadden-riscv to reach out to you via your personal email.  When you get your IITM email, we'll get you authorized to things.  Please let me know.

Thanks for getting started on this!!!

**Comment by JAYANTH-IITM** [comment]
Updates , 

@jjscheel  No progress since last call , was working on Zce ACT

**Comment by JAYANTH-IITM** [comment]
Updates from IIT-M , 
 
Made changes suggested by bill here (https://github.com/saravanakumarsastra/sail-riscv/tree/billmc-comments) in my github repo and shared it for review 

Github link : https://github.com/JAYANTH-IITM/sail-riscv

**Comment by IC41558-IITM** [comment]
Hi @billmcspadden-riscv  , 
Following up on my progress , were you able review the updated PLIC sail model.

**Comment by JAYANTH-IITM** [comment]
Status Update :

Sent for the suggested changes for review to Bill.

**Comment by JAYANTH-IITM** [comment]
Status Update :
Working on resolving error during compilation

**Comment by jjscheel** [comment]
Jayanth has been assigned to other gaps for now.  Will unblock when Jayanth has time again.

**Comment by jjscheel** [comment]
Tim Hutt in the Sail community built a PLIC interrupt controller [link](https://github.com/Timmmm/sail-riscv/tree/user/timh/plic).

Closing.
